cpp/include/cuspatial/point_in_polygon.hpp

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <memory>
+#include <cudf/column/column.hpp>
+#include <cudf/column/column_view.hpp>
+#include "cudf/types.hpp"
+
+namespace cuspatial {
+
+/**
+ * @brief Point-in-Polygon (PIP) tests between a column of points and a
+ *        column of polygons
+ *
+ * Performs collision detection for each test point against at most 32 polygons. Polygons are
+ * a collection of one or more rings. Rings are a collection of three or more vertices.
+ *
+ * @param[in] test_points_x:     x component of target points
+ * @param[in] test_points_y:     y component of target points
+ * @param[in] poly_offsets:      beginning index of the first ring in each polygon
+ * @param[in] poly_ring_offsets: beginning index of the first point in each ring
+ * @param[in] poly_points_x:     x component of polygon points
+ * @param[in] poly_points_y:     y component of polygon points
+ *
+ * @returns A column of cudf::size_type containing one element per input point. Each bit
+ * represents a hit or miss for each of the input polygons in least-significant-bit order.
+ * i.e. `output[3] & 0x0010` indicates a hit or miss for the 3rd point against the 2nd polygon.
```

Comment:
0x0010 is the hexadecimal number 16, so that represents a hit against the 17th polygon, if bit 0 is the first polygon.

```suggestion
 * i.e. `output[3] & 0x0010` indicates a hit or miss for the 3rd point against the 17th polygon.
```

Reply:
Good catch. `0x0010` should have been `0b0010`.